### PR TITLE
Fire binding

### LINKED ISSUE #3534: Remove the ReactiveBindingHolder and its extensions
### The problem

We introduced the `Bind` extension as a way to make our code less complex. However, after the `AsDriver` refactor, the extension lost its meaning and now serves just as another way of doing things, increasing complexity.

### The solution

Let's completely remove this extension and instead favor using the well known formula for Reactive for subscriptions:

```csharp
ViewModel.SomeTextObservable.
    .Subscribe(someTextView.Rx().Text())
    .DisposedBy(DisposeBag);
```

### PR DESCRIPTION
**Pull request by Odrakir**
## What's this?
Removes the old `IReactiveBindinHolder` and its extensions. This is more code, but less magic. I think the main problem with the bindings were the multiple overrides of `Bind`, you never knew what it'd end up doing.

Closes #3534 

## Why do we want this?
Less magic, easier to understand code.

## How is it done?
I removed the extensions and then replaced all `Bind` calls with regular `Subscribe` and `DisposedBy`.

### Side effects
This is a lot of mechanical changes, so there might be some mistakes... We should do some testing.

## Review hints
There's some small commits with needed changes and then there's the big `Remove usage of old bind methods` with all the binding replacements, probably not worth it to look at every line of it, bu if you have the time...

Also some @toggl/androids should take a look at `Remove IReactiveBindingHolder from Giskard`, please.

## :squid: Permissions
If you dare... fine by me.

### CODE REVIEW

File: Toggl.Daneel/Extensions/Reactive/UIViewExtensions.cs [hunk 1]
```diff
@@ -66,5 +67,13 @@ public static Action<float> AnimatedAlpha(this IReactive<UIView> reactive)
                 );
             };
 
+        public static IDisposable BindAction(this IReactive<UIView> reactive, UIAction action)
+        {
+            return Observable.Using(
+                    () => { return action.Enabled.Subscribe(e => { reactive.Base.UserInteractionEnabled = e; }); },
```
Comment (simonrozsival) [inline Toggl.Daneel/Extensions/Reactive/UIViewExtensions.cs]: I know you like `return`s but I think here they are redundant:
```suggestion
                    () => action.Enabled.Subscribe(e => { reactive.Base.UserInteractionEnabled = e; }),
```

File: Toggl.Daneel/Extensions/Reactive/UIViewExtensions.cs [hunk 2]
```diff
@@ -66,5 +67,13 @@ public static Action<float> AnimatedAlpha(this IReactive<UIView> reactive)
...
+                    _ => { return reactive.Base.Rx().Tap(); }
```
Comment (simonrozsival) [inline Toggl.Daneel/Extensions/Reactive/UIViewExtensions.cs]:
```suggestion
                    _ => reactive.Base.Rx().Tap()
```

File: Toggl.Daneel/Extensions/Reactive/UIButtonExtensions.cs [hunk 3]
```diff
@@ -32,5 +40,26 @@ public static Action<string> AnimatedTitle(this IReactive<UIButton> reactive)
...
+                    eventObservable = reactive.Base.Rx().Tap();
+                    break;
+                case ButtonEventType.LongPress:
+                    throw new ArgumentException("Event type not implemented");
+                    //eventObservable = reactive.Base.Rx().LongPress();
+                    break;
+            }
+
+            return Observable.Using(
+                    () => { return action.Enabled.Subscribe(e => { reactive.Base.Enabled = e; }); },
```
Comment (simonrozsival) [inline Toggl.Daneel/Extensions/Reactive/UIButtonExtensions.cs]:
```suggestion
                    () => action.Enabled.Subscribe(e => { reactive.Base.Enabled = e; }),
```

File: Toggl.Daneel/Extensions/Reactive/UIButtonExtensions.cs [hunk 4]
```diff
@@ -32,5 +40,26 @@ public static Action<string> AnimatedTitle(this IReactive<UIButton> reactive)
...
+                    _ => { return eventObservable; }
```
Comment (simonrozsival) [inline Toggl.Daneel/Extensions/Reactive/UIButtonExtensions.cs]:
```suggestion
                    _ => eventObservable
```

File: Toggl.Giskard/Extensions/Reactive/ViewExtensions.cs [hunk 5]
```diff
@@ -19,5 +25,26 @@ public static Action<bool> Enabled(this IReactive<View> reactive)
...
+                    eventObservable = reactive.Base.Rx().Tap();
+                    break;
+                case ButtonEventType.LongPress:
+                    throw new ArgumentException("Event type not implemented");
+                    //eventObservable = reactive.Base.Rx().LongPress();
+                    break;
+            }
+
+            return Observable.Using(
+                    () => { return action.Enabled.Subscribe(e => { reactive.Base.Enabled = e; }); },
```
Comment (simonrozsival) [inline Toggl.Giskard/Extensions/Reactive/ViewExtensions.cs]:
```suggestion
                    () => action.Enabled.Subscribe(e => { reactive.Base.Enabled = e; }),
```

File: Toggl.Giskard/Extensions/Reactive/ViewExtensions.cs [hunk 6]
```diff
@@ -19,5 +25,26 @@ public static Action<bool> Enabled(this IReactive<View> reactive)
...
+                    _ => { return eventObservable; }
```
Comment (simonrozsival) [inline Toggl.Giskard/Extensions/Reactive/ViewExtensions.cs]:
```suggestion
                    _ => eventObservable
```

File: Toggl.Daneel/ViewControllers/LoginViewController.cs [hunk 7]
```diff
@@ -46,41 +48,101 @@ public override void ViewDidLoad()
...
+                .Select(loginButtonTitleColor)
+                .Subscribe(LoginButton.Rx().TitleColor())
+                .DisposedBy(DisposeBag);
 
-                if (shakeTargets.HasFlag(LoginViewModel.ShakeTargets.Password))
-                    PasswordTextField.Shake();
-            });
+            //Animation
+            ViewModel.Shake
+                .Subscribe(shakeTargets =>
```
Comment (willsb) [inline Toggl.Daneel/ViewControllers/LoginViewController.cs]: I'd maybe split this into two bindings?

```csharp
ViewModel.Shake
    .Select(shakeTargets => shakeTargets.HasFlag(LoginViewModel.ShakeTargets.Email)
    .VoidSubscribe(() => EmailTextField.Shake())
    .DisposedBy(DisposeBag);
```

File: Toggl.Daneel/ViewControllers/Settings/UpcomingEventsNotificationSettingsViewController.cs [hunk 8]
```diff
@@ -27,8 +28,13 @@ public override void ViewDidLoad()
...
             TableView.SelectRow(NSIndexPath.FromRowSection(ViewModel.SelectedOptionIndex, 0), false, UITableViewScrollPosition.None);
 
-            this.Bind(dataSource.SelectedOptionChanged, ViewModel.SelectOption);
-            this.Bind(CloseButton.Rx().Tap(), ViewModel.Close);
+            dataSource.SelectedOptionChanged
+                .Subscribe(ViewModel.SelectOption.Inputs)
+                .DisposedBy(DisposeBag);
+
+            CloseButton.Rx()
+                .BindAction(ViewModel.Close)
```
Comment (willsb) [inline Toggl.Daneel/ViewControllers/Settings/UpcomingEventsNotificationSettingsViewController.cs]: This (and some others) will clash with the Close action PR. Let's just be aware of that

File: Toggl.Daneel/ViewControllers/LoginViewController.cs (hunk 7 above)
Comment (Odrakir) [reply Toggl.Daneel/ViewControllers/LoginViewController.cs]: For now I'm not changing the old way of doing things, I'm just removing the binding syntax. There'll be another PR to fix some things I found, we can talk about this then.

File: Toggl.Daneel/ViewControllers/Settings/UpcomingEventsNotificationSettingsViewController.cs (hunk 8 above)
Comment (Odrakir) [reply Toggl.Daneel/ViewControllers/Settings/UpcomingEventsNotificationSettingsViewController.cs]: What PR are you talking about?

**Comment by Odrakir** [comment]
@daividssilverio In my opinion the code we save is not worth the headache. This is a bit more of code, but it's clear on what it does. I would leave it like this for now.

File: Toggl.Daneel/ViewControllers/Settings/UpcomingEventsNotificationSettingsViewController.cs (hunk 8 above)
Comment (Odrakir) [reply Toggl.Daneel/ViewControllers/Settings/UpcomingEventsNotificationSettingsViewController.cs]: Never mind, I just hit the conflicts.

File: Toggl.Daneel/ViewControllers/Calendar/CalendarPermissionDeniedViewController.cs [hunk 9]
```diff
@@ -5,8 +5,12 @@
 using CoreGraphics;
 using UIKit;
 using Toggl.Daneel.Extensions.Reactive;
+<<<<<<< HEAD
```
Comment (willsb) [inline Toggl.Daneel/ViewControllers/Calendar/CalendarPermissionDeniedViewController.cs]: 👀

Comment (Odrakir) [reply Toggl.Daneel/ViewControllers/Calendar/CalendarPermissionDeniedViewController.cs]: Rider hiding the imports when it shouldn't... 🙄

**Pull request merged**